index.js

```diff
@@ -3,8 +3,73 @@
 const Scm = require('screwdriver-scm-base');
 const hoek = require('hoek');
 const url = require('url');
+const request = require('request');
+const schema = require('screwdriver-data-schema');
+const API_URL = 'https://api.bitbucket.org/2.0/repositories/';
+const MATCH_COMPONENT_HOST_HTTPS = 2;
+const MATCH_COMPONENT_HOST_SSH = 3;
+const MATCH_COMPONENT_USER = 4;
+const MATCH_COMPONENT_REPO = 5;
+const MATCH_COMPONENT_BRANCH = 6;
+
+/**
+ * Get repo information
+ * @method  getRepoInfo
+ * @param   {String}    checkoutUrl     The url to check out repo
+ * @return  {Object}                    An object with host, repo, branch, and username
+ */
+function getRepoInfo(checkoutUrl) {
+    const regex = schema.config.regex.CHECKOUT_URL;
+    const matched = regex.exec(checkoutUrl);
+    const result = {
+        username: matched[MATCH_COMPONENT_USER],
+        repo: matched[MATCH_COMPONENT_REPO],
+        branch: matched[MATCH_COMPONENT_BRANCH].slice(1)
+    };
+    let host;
+
+    if (checkoutUrl.indexOf('https') > -1) {
```

Comment:
Better Regex:
```
/^(?:(?:https?|git):\/\/)?(?:[^@]+@)?([^/:]+)(?:/|:)([^\/]+)\/([^.]+)(?:\.git)?(#.+)?$/
```

Matches:
 - https://stjohnjohnson@bitbucket.org/stjohnjohnson/foobar
 - git@bitbucket.org:stjohnjohnson/foobar.git
 - git@github.com:screwdriver-cd/data-schema.git
 - https://github.com/screwdriver-cd/data-schema.git

With result:
 - 0 is hostname
 - 1 is org
 - 2 is repo
 - 3 is branch